### PR TITLE
matplotlib as optional dependency

### PR DESCRIPTION
**Pull request by FranzForstmayr**
This PR makes matplotlib an optional dependency again and adds a testsetup which tries to test the most basic network commands without matplotlib installed.

### CODE REVIEW

File: skrf/tests/test_vectorfitting.py [hunk 1]
```diff
@@ -101,9 +105,9 @@ def test_read_write_npz(self):
         self.assertTrue(np.allclose(vf.proportional_coeff, vf2.proportional_coeff))
         self.assertTrue(np.allclose(vf.constant_coeff, vf2.constant_coeff))
 
+    @pytest.mark.skipif("matplotlib" in sys.modules, reason="Raise Error only if matplotlib is not installed.")
```
Comment (mhuser) [inline skrf/tests/test_vectorfitting.py:108]: The test is usually not skipped because matplotlib is no more in the dependencies, is it like that ?

**Review by mhuser — APPROVED** [review]
Excellent, you definitively are a decorator master !

File: skrf/tests/test_vectorfitting.py (hunk 1 above)
Comment (FranzForstmayr) [reply skrf/tests/test_vectorfitting.py:108]: This test is skipped for the default test runs (py37 to py311) as matplotlib is installed and no exception is raised.
The test is executed in the minimal setup and asserts the RuntimeError

Comment (mhuser) [reply skrf/tests/test_vectorfitting.py:108]: Great!

File: skrf/vectorFitting.py [hunk 2]
```diff
@@ -1388,9 +1369,9 @@ def get_model_response(self, i: int, j: int, freqs: Any = None) -> np.ndarray:
                 resp += residues[i] / (s - pole) + np.conjugate(residues[i]) / (s - np.conjugate(pole))
         return resp
 
-    @check_plotting
+    @axis_kwarg
     def plot(self, component: str, i: int = -1, j: int = -1, freqs: Any = None,
-             parameter: str = 's', ax: mplt.Axes = None) -> mplt.Axes:
+             parameter: str = 's', *,ax = None):
```
[the remark this answers is not in the document]
Comment (FranzForstmayr) [reply skrf/vectorFitting.py]: One should mention, that with this change `ax` is a keyword-only property. I hope this is fine @Vinc0110?

**Review by mhuser — APPROVED** [review]

File: skrf/vectorFitting.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Vinc0110) [reply skrf/vectorFitting.py]: It should be fine. I did not know this ugly syntax, but it seems to be according to [PEP 3102](https://peps.python.org/pep-3102/).

**Review by jhillairet — APPROVED** [review]
Nice and clean. Eventually, a comment could be added in the tox.ini to remember why this minimal env exists.

**Pull request merged**